concent_api/core/views.py

```diff
@@ -1,207 +1,308 @@
-import json
 import datetime
+from base64                         import b64decode
 
 from django.http                    import HttpResponse
 from django.views.decorators.http   import require_POST
 from django.utils                   import timezone
 from django.conf                    import settings
 
-from utils.api_view                 import api_view, Http400
-from .models                        import Message, MessageStatus
+from golem_messages.message         import MessageAckReportComputedTask
+from golem_messages.message         import MessageForceReportComputedTask
+from golem_messages.message         import MessageRejectReportComputedTask
+from golem_messages.message         import MessageCannotComputeTask
+from golem_messages.message         import MessageTaskFailure
+from golem_messages.message         import MessageTaskToCompute
+from golem_messages.message         import MessageVerdictReportComputedTask
+from golem_messages.shortcuts       import dump
+from golem_messages.shortcuts       import load
+
+from utils.api_view                 import api_view
+from utils.api_view                 import Http400
+from .models                        import Message
+from .models                        import MessageStatus
 
 
 @api_view
 @require_POST
 def send(request, message):
-    data = message
-    validate_message(data)
-
-    if data['type'] == "MessageForceReportComputedTask":
-        validate_message_task_to_compute(data['message_task_to_compute'])
-
-        if Message.objects.filter(task_id = data['message_task_to_compute']['task_id']).exists():
-            raise Http400("'ForceReportComputedTask' is already being processed for this task.")
+    client_public_key = decode_client_public_key(request)
+    if isinstance(message, MessageForceReportComputedTask):
+        loaded_message = load(
+            message.message_task_to_compute,
+            settings.CONCENT_PRIVATE_KEY,
+            client_public_key
+        )
+        validate_golem_message_task_to_compute(loaded_message)
+        if Message.objects.filter(task_id = loaded_message.task_id).exists():
+            raise Http400(message.__class__.__name__, " is already being processed for this task.")
```

Comment:
This will not construct the message correctly. You'll get something like
`('MessageForceReportComputedTask', ' is already being processed for this task.')`,
not
`MessageForceReportComputedTask is already being processed for this task.`